builder/dockerfile/internals.go

```diff
@@ -489,8 +489,10 @@ func hostConfigFromOptions(options *types.ImageBuildOptions, isWCOW bool) *conta
 	// For WCOW, the default of 20GB hard-coded in the platform
 	// is too small for builder scenarios where many users are
 	// using RUN statements to install large amounts of data.
-	// Use 127GB as that's the default size of a VHD in Hyper-V.
-	if isWCOW {
+	// If no custom size is set, use 127GB for containers that
+	// are started during build, as that's the default size of
+	// a VHD in Hyper-V.
+	if isWCOW && (len(hc.StorageOpt) == 0 || len(hc.StorageOpt) > 0 && hc.StorageOpt["size"] == "") {
```

Comment:
```suggestion
if isWCOW && hc.StorageOpt["size"] == {""
```

Unless I'm missing something this seems like what we'd want.

Reply:
Forgot about this one; yes, looks like all the extra checks are not needed